File: app/src/main/java/com/eagskunst/emmanuel/gamingnews/Objects/LoadUrls.java
package com.eagskunst.emmanuel.gamingnews.Objects;

import com.eagskunst.emmanuel.gamingnews.Models.Categories;
import com.google.gson.Gson;

import java.io.IOException;
import java.io.InputStream;

public class LoadUrls {
    private String language;
    private InputStream jsonFile;
    private String[] allUrls;
    private String[] ps4Urls;
    private String[] xboxOUrls;
    private String[] switchUrls;
    private String[] pcUrls;

    public LoadUrls(String language, InputStream jsonFile){
        this.language = language;
        this.jsonFile = jsonFile;
    }
    private Categories[] loadCategories(){
        Gson gson = new Gson();
        return gson.fromJson(loadJSONFromAsset(),Categories[].class);
    }
    private Categories getLocalCategory(){
        Categories[] categories = loadCategories();
        Categories category = null;
        for(Categories cat:categories){
            if(cat.getLanguage().equals(language)){
                category = cat;
                break;
            }
            else if(cat.getLanguage().equals("en") && category == null){
                category = cat;
            }
        }
        return category;
    }

    public void setUrls(){
        Categories category = getLocalCategory();
        allUrls = category.getAll_urls();
        ps4Urls = category.getPs4_urls();
        xboxOUrls = category.getXboxO_urls();
        switchUrls = category.getSwitch_urls();
        pcUrls = category.getPc_urls();
    }

    private String loadJSONFromAsset() {
        String json = null;
        try {
            InputStream is = jsonFile;
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            is.close();
            json = new String(buffer, "UTF-8");
            return json;
        } catch (IOException ex) {
            ex.printStackTrace();
        }
        return json;
    }


    public String[] getAllUrls() {
        return allUrls;
    }

    public String[] getPs4Urls() {
        return ps4Urls;
    }

    public String[] getXboxOUrls() {
        return xboxOUrls;
    }

    public String[] getSwitchUrls() {
        return switchUrls;
    }

    public String[] getPcUrls() {
        return pcUrls;
    }

}

File: app/src/main/java/com/eagskunst/emmanuel/gamingnews/objects/LoadUrls.kt
package com.eagskunst.emmanuel.gamingnews.objects

import com.eagskunst.emmanuel.gamingnews.models.Categories
import com.google.gson.Gson
import java.io.IOException
import java.io.InputStream

class LoadUrls(private val language: String, private val jsonFile: InputStream) {
    var allUrls: Array<String>? = null
    var ps4Urls: Array<String>? = null
    var xboxOUrls: Array<String>? = null
    var switchUrls: Array<String>? = null
    var pcUrls: Array<String>? = null

    private fun loadCategories(): Array<Categories> {
        val gson = Gson()
        return gson.fromJson(loadJSONFromAsset(), Array<Categories>::class.java)
    }

    private fun getLocalCategory(): Categories? {
        val categories = loadCategories()
        var category: Categories? = null
        for (cat in categories) {
            if (cat.language == language) {
                category = cat
                break
            } else if (cat.language == "en" && category == null) {
                category = cat
            }
        }
        return category
    }

    fun setUrls() {
        val category = getLocalCategory()
        allUrls = category?.all_urls
        ps4Urls = category?.ps4_urls
        xboxOUrls = category?.xboxO_urls
        switchUrls = category?.switch_urls
        pcUrls = category?.pc_urls
    }

    private fun loadJSONFromAsset(): String? {
        var json: String? = null
        try {
            val size = jsonFile.available()
            val buffer = ByteArray(size)
            jsonFile.read(buffer)
            jsonFile.close()
            json = String(buffer, Charsets.UTF_8)
            return json
        } catch (ex: IOException) {
            ex.printStackTrace()
        }
        return json
    }
}
